### PR TITLE
Feature#4 router

### PR DESCRIPTION
**Pull request by tania833**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
 
 ReactDOM.render(
   <React.StrictMode>
-    <App />
+    <BrowserRouter>
+      <App />
+    </BrowserRouter>
   </React.StrictMode>,
-  document.getElementById('root')
+  document.getElementById("root"),
```
Comment (SergeOlabin) [inline src/index.tsx:14]: let's stick with single-quotes

File: src/App.tsx [hunk 2]
```diff
@@ -1,25 +1,30 @@
...
-import './App.css';
- 
-function App() {
+import React, { Suspense } from "react";
+import "./App.css";
+import logo from "./logo.svg";
+import { RenderRoutes } from "./navigation/Router";
+import ROUTER_CONFIG from "./navigation/router-config";
+
+function App(): JSX.Element {
```
Comment (SergeOlabin) [inline src/App.tsx:7]:
```suggestion
const App: React.FC = () => {
```

File: src/navigation/Router.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+export const RouteWithSubRoutes: React.FC<IRouterConfig> = (
+  route: IRouterConfig,
+) => {
+  const { routes } = route;
+  if (routes === undefined) {
+    const routeWithoutRoutes = route as IRouterConfigWithoutRoutes;
+    return (
+      <Route path={route.path} exact={route.exact}>
+        <routeWithoutRoutes.component />
```
Comment (SergeOlabin) [inline src/navigation/Router.tsx:13]: 👍

File: src/navigation/Router.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      </Route>
+    );
+  }
+  return (
+    <Route
+      path={route.path}
+      exact={route.exact}
+      render={() => <route.component routes={routes} />}
```
Comment (SergeOlabin) [inline src/navigation/Router.tsx:21]: can do the same as on line 12:
```
<Route path={route.path} exact={route.exact}>
        <routeWithoutRoutes.component />
</Route>
```
?

File: src/navigation/types.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    component: React.ComponentType<{
+        routes: IRouterConfig[]
+    }>,
+    path: string,
+    key?: string,
+    exact?: boolean,
+    routes?: IRouterConfig[],
+}
+
+export interface IRouterConfigWithoutRoutes {
```
Comment (SergeOlabin) [inline src/navigation/types.ts:13]: why do we need 2 router configs?
it should be possible to have just 1

File: src/navigation/types.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+
```
Comment (SergeOlabin) [inline src/navigation/types.ts:1]: empty lines

**Review by SergeOlabin — COMMENTED** [review]
pls check .gitignore if it contains `package-lock.json` it should not be in the  changes

File: src/navigation/Router.tsx (hunk 4 above)
Comment (tania833) [reply src/navigation/Router.tsx:21]: Done

File: src/navigation/types.ts (hunk 5 above)
Comment (tania833) [reply src/navigation/types.ts:13]: Have no idea how to omit it

File: src/navigation/types.ts (hunk 6 above)
Comment (tania833) [reply src/navigation/types.ts]: deleted

File: src/App.tsx (hunk 2 above)
Comment (tania833) [reply src/App.tsx]: Done

File: src/index.tsx (hunk 1 above)
Comment (tania833) [reply src/index.tsx]: Ok, it would be great if quotes were automatically added with Eslint

File: src/navigation/Router.tsx [hunk 7]
```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import { Route, Switch } from "react-router-dom";
+import { IRouterConfig } from "./types";
+
+export const RouteWithSubRoutes: React.FC<IRouterConfig> = (props) => {
+  const { routes, path, exact } = props;
+
+  if (routes === undefined) {
```
Comment (SergeOlabin) [inline src/navigation/Router.tsx:8]:
```suggestion
  if (!routes) {
```

**Pull request merged**